odoo/tools/config.py

```diff
@@ -530,7 +530,7 @@ def die(cond, msg):
         self.options['translate_modules'] = opt.translate_modules and [m.strip() for m in opt.translate_modules.split(',')] or ['all']
         self.options['translate_modules'].sort()
 
-        dev_split = [s.strip() for s in opt.dev_mode.split(',')] if opt.dev_mode else []
+        dev_split = [s.strip() for s in opt.dev_mode.split(',')] if opt.dev_mode else [self.options['dev_mode']] or []
```

Comment:
I think that this is not correct:

### Save format
I'm not very familiar with the config file parsing but I don't think it is possible to get the array directly from the config. The `addons_path` and other comma separated elements are stored as a comma separated string, I would expect the `dev_mode` to be stored the same way. 
Also, if i remove the dev_mode from the blacklist for save, I would expect it to work correctly. Right now it would look like 
`dev_mode = ['xml', 'qweb']`

And be loaded as
`["['xml', 'qweb']"]` 

IMHO The good solution would be to fallback on options before parsing it, and don't split it in options but in conf

### Lets make it overridable with command line

Also, checking only if command line option is falsy is not enough if wa wan to be able to override it from command line with an empty value, lalsy vs None.


### suggestion

```patch
--- a/odoo/tools/config.py
+++ b/odoo/tools/config.py
@@ -530,9 +530,6 @@ class configmanager(object):
         self.options['translate_modules'] = opt.translate_modules and [m.strip() for m in opt.translate_modules.split(',')] or ['all']
         self.options['translate_modules'].sort()
 
-        dev_split = [s.strip() for s in opt.dev_mode.split(',')] if opt.dev_mode else []
-        self.options['dev_mode'] = dev_split + (['reload', 'qweb', 'xml'] if 'all' in dev_split else [])
-
         if opt.pg_path:
             self.options['pg_path'] = opt.pg_path
 
@@ -545,6 +542,10 @@ class configmanager(object):
         for key in ['data_dir', 'logfile', 'pidfile', 'test_file', 'screencasts', 'screenshots', 'pg_path', 'translate_out', 'translate_in', 'geoip_city_db', 'geoip_country_db']:
             self.options[key] = self._normalize(self.options[key])
 
+        dev_split = [s.strip() for s in (self.options['dev_mode'] or '').split(',') if s]
+        dev_mode = dev_split + (['reload', 'qweb', 'xml'] if 'all' in dev_split else [])
+        conf.dev_mode = dev_mode
+
         conf.addons_paths = self.options['addons_path'].split(',')
 
         conf.server_wide_modules = [
```

It was quickly tested but I'm not sure that my suggestion is the right one, to discuss and test.

Reply:
thanks @Xavier-Do ,
updated PR with suggested changes, there were one exception happening as dev_mode key was missing in the self.options , so assigned the value of dev_mode variable to self.options also.